packages/common/src/api.ts

```diff
@@ -17,4 +19,37 @@ export function getHubApiUrl(requestOptions: IRequestOptions): string {
   
 }
 
+export function getFromHubAPI(id:string, hubRequestOptions?: IHubRequestOptions): Promise<any> {
+  let hubUrl = "https://hub.arcgis.com/api";
+  let headers:any = {};
+
+  if (hubRequestOptions !== undefined) {
+    hubUrl = hubRequestOptions.hubApiUrl;
+    headers['Authorization'] =  hubRequestOptions.authentication.token;
+  }
+  
+  const url = `${hubUrl}/v3/datasets/${id}`;
+  const opts = {
+    method: "GET",
+    mode: "cors",
+    headers
+  } as RequestInit;
 
+  // TODO: handle Hub missing content by ID
+  return fetch(url, opts)
```

Comment:
We should consider having a `hubFetch` general purpose wrapper to centralize `fetch` similar to rest-js...

Reply:
yes. This was _quicky + dirty_. Should we just use rest-js version of fetch?

Reply:
If you mean `requestOptions.fetch`, then yes, we should probably favor that.